kbase-extension/static/kbase/js/common/parameterSpec.js

```diff
@@ -231,23 +197,18 @@ define([], () => {
             const defaultValues = spec.default_values;
             // No default value and not required? null value
 
-            // special special cases.
-            switch (spec.field_type) {
-                case 'checkbox':
-                    /*
-                     * handle the special case of a checkbox with no or empty
-                     * default value. It will promote to the "unchecked value"
-                     * TODO: more cases of bad default value? Or a generic
-                     * default value validator?
-                     */
-                    if (!defaultValues || defaultValues.length === 0) {
-                        return spec.checkbox_options.unchecked_value;
-                    } else {
-                        return coerceToIntBoolean(defaultValues[0]);
-                    }
-                case 'custom_textsubdata':
-                    if (!defaultValues) {
-                    }
+            if (spec.field_type === 'checkbox') {
+                /*
+                 * handle the special case of a checkbox with no or empty
+                 * default value. It will promote to the "unchecked value"
+                 * TODO: more cases of bad default value? Or a generic
+                 * default value validator?
+                 */
+                if (!defaultValues || defaultValues.length === 0) {
+                    return spec.checkbox_options.unchecked_value;
+                } else {
+                    return coerceToIntBoolean(defaultValues[0]);
+                }
```

Comment:
I should have removed this unneeded `else` when I got rid of the `switch` statement, but oh well.